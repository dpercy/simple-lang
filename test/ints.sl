123
4294967296 ; 2^32
(+ 1 4294967296)
18446744073709551616 ; 2^64
(+ 1 18446744073709551616)
(def (fact n)
  (match (< n 2)
    [#true n]
    [#false (* n (fact (- n 1)))]))
(fact 50)
(/ 3 0)
(- 23 100)
-5
-0
(- 1 1)
(- 0 0)
(= 0 -0)
(/ 3 -0)
