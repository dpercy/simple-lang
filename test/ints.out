123
4294967296
4294967297
18446744073709551616
18446744073709551617
30414093201713378043612608166064768844377641568960512000000000000
(error "/: division by zero")
-77
-5
0
0
0
#true
(error "/: division by zero")
